storage/bucket.go

```diff
@@ -1367,6 +1376,19 @@ func (b *BucketHandle) Objects(ctx context.Context, q *Query) *ObjectIterator {
 	return it
 }
 
+// Retryer returns a bucket handle that is configured with custom retry
+// behavior as specified by the options that are passed to it. All operations
+// on the new handle will use the customized retry configuration.
+func (b *BucketHandle) Retryer(opts ...RetryOption) *BucketHandle {
```

Comment:
This looks like the right surface.

I realized we probably also need to account for resolving the options between BucketHandle and ObjectHandle-- for example, if the user calls `client.Bucket(...).Retryer(...).Object(...)`, I guess the retry config should be inherited from the bucket. We should probably discuss what the resolution strategy should look like.